walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -25,59 +25,59 @@
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
 
     // Handle edge cases: when there are no attendees & the meeting duration is longer than a day
-    if(request.getAttendees().isEmpty()) {
+    if(request.getAttendees().isEmpty() && request.getOptionalAttendees().isEmpty()) {
         return Arrays.asList(TimeRange.WHOLE_DAY);
     } else if(request.getDuration() > TimeRange.END_OF_DAY) {
         return Arrays.asList();
     }
     
-    // List of TimeRanges that will not work for meeting attendees
-    Collection<TimeRange> blockedTimes = new ArrayList();
+    Collection<String> allAttendees = new ArrayList();
+    allAttendees.addAll(request.getAttendees());
+    allAttendees.addAll(request.getOptionalAttendees());
     // List of TimeRanges that will not work for meeting attendees + optional attendees
-    Collection<TimeRange> optionalBlockedTimes = new ArrayList();
-    
-    createListOfBlockedTimes(events, request, blockedTimes);
-    createListOfBlockedTimes(events, request, optionalBlockedTimes);
-
-    // List of TimeRanges that will work for meeting attendees
-    Collection<TimeRange> availableTimes = new ArrayList();
+    Collection<TimeRange> optionalBlockedTimes = createListOfBlockedTimes(events, allAttendees);
     // List of TimeRanges that will work for meeting attendees + optional attendees
-    Collection<TimeRange> optionalAvailableTimes = new ArrayList();
+    Collection<TimeRange> optionalAvailableTimes = createListOfAvailableTimes(events, request, optionalBlockedTimes);
 
-    createListOfAvailableTimes(events, request, blockedTimes, availableTimes);
-    createListOfAvailableTimes(events, request, optionalBlockedTimes, optionalAvailableTimes);
-
-    // Only return the List that works for optional attendees if there is one or more TimeRange
-    if (optionalAvailableTimes.size() > 0) {
+    // Return the List that works for optional attendees if there is one or more TimeRange or no attendees
+    if (optionalAvailableTimes.size() > 0 || (optionalAvailableTimes.isEmpty() && request.getAttendees().isEmpty())) {
```

Comment:
i think you can make the first condition `!optionalAvailableTimes.isEmpty()`

Reply:
Done!